### ISSUE TITLE
Q: Are adaptive payments deprecated?

### ISSUE THREAD

**Issue by dcworldwide** [Ideame/paypal-adaptive-sdk-nodejs#16]
Has paypal deprecated adaptive payments?

I'm wondering if this is still the best api to use when facilitating a peer to peer paypal transaction, i.e. Where your app brokers a batch transaction between a payer and multiple payees.

I'm a paypal noob so if you could share your thoughts that would be really helpful. If adaptive payments are still relevant I will be using this lib and will contribute in anyway I can.

**Comment by mjmasn** [comment]
(Just a PayPal customer here so this may not be correct)

It looks like you can still access the 'Create New App' form for legacy applications at https://www.paypal-apps.com/user/my-account/applications. However PayPal have in the past said they will deprecate these APIs when the newer REST APIs have feature parity. So even if you do get approved you'll have to deal with that eventually. 

We have 2 existing AP apps and if we submit any further requests or questions online they tend to go unanswered until we phone them repeatedly. 

So in conclusion, you should probably ring PayPal and ask to speak to someone who knows about AP for a conclusive answer (this may take a few attempts!).

**Comment by dcworldwide** [comment]
Ah, I was afraid of this. It mirrors my history with other large api service providers. 
Thanks for your thoughts, they're really helpful. 

I've submitted my AP app, and have been auto granted sandbox permissions. My sandbox testing works great with this lib. But as you said, my big concern is getting production approval....From what you've said it sounds like they are going to give me a very difficult time about it :(

**Comment by mjmasn** [comment]
Good luck!

**Comment by rubenjimenez** [comment]
HI,

We're exactly in the same point that @dcworldwide. We're starting to develop a solution for creating payments among different users and taking our commission in the middle. According to what we've been reading, Adaptive Payments is what we need but it seems like it is a little bit deprecated.

Reading about the new REST API, we haven't found support for what we want to accomplish. There's MassPay, but I think that doesn't allow us to get the same payment flow like chained payments.

So, do you recommend to use Adaptive Payments? @dcworldwide, what about your production approval? Did you get it successfully?

Thanks!

**Comment by aaadarshbisht** [comment]
@rubenjimenez  Hi ,I am also facing the same problem ,I want users to transfer money b/w them,
Did you find solution to this problem,Thanks?

**Comment by rubenjimenez** [comment]
Hi @aaadarshbisht,

We've finally used this Node package and it does what it says. We haven't had any kind of problem in stage environments. We've not tested it out in production yet as we're waiting for PayPal approval for starting to run it with real payments.

I'll update this thread when we get our approval.

Regards!

**Comment by dcworldwide** [comment]
Please do +1

**Comment by kanedaki** [comment]
Any news on this? I'm about to choose between AP or... or rethink my app, so any help here would be much appreciated

Thanks

**Comment by rubenjimenez** [comment]
Although we did the development, we didn't never test it in production env due to PayPal didn't allow it. It's true PayPal told our client that the problem was the business model and some things beyond the development itself. But, as I said, I don't know how it works in production. Our client hopes PayPal will finally allow it soon.

I'll update if it is the case.

**Comment by mjmasn** [comment]
@rubenjimenez we initially had a response saying our business model was an issue (IIRC having 50/50 split between us and a third-party) but this was resolved by stating that we would take responsibility for all refunds - not sure if that will help in your case?

I thought I read a few weeks ago that they were no longer accepting applications for Adaptive Payments - is that not the case? That said, it's entirely possible for this to be a new rule without the relevant PayPal dept. knowing about it if my experience of their support team is anything to go by.

Good luck!

**Comment by rubenjimenez** [comment]
@mjmasn I'll tell my client about the refunds, it may help them to get the approval.

If PayPal is not accepting applications for Adaptive Payments anymore (I don't know, I haven't tried to get the approval some months ago), my client has a problem =) I'll tell them as well and let's see what we'll do.

Anyway, is there any new way to make split payments with PayPal beyond Adaptive Payments?

**Comment by dcworldwide** [comment]
Good question, is there an alternative?

From what I understand, APs is the only way to facilitate the transfer of money between users (i.e. 1:1 or 1:M)....

**Comment by vincelwt** [comment]
@rubenjimenez Did your client manage to get the approval? I'm in the same situation but from what I see there isn't any clear alternative

**Comment by rubenjimenez** [comment]
Hi @vincelwt,

No, they didn't. Currently, everything is stopped and they are thinking on another different solution.

**Comment by dcworldwide** [comment]
From what I understand, existing PayPal Adaptive payment customers can continue using the solution until further notice. There is no end date for the service. New approvals likely won't get accepted.

Can someone else confirm?

**Comment by rustemkk** [comment]
Hello everyone! 
@dcworldwide @rubenjimenez @vincelwt @mjmasn @kanedaki @aaadarshbisht
Can you please provide any actual information (2019) on how to make a new app using paypal adaptive payments or analogue (user1 sends money to user2 and our app is getting part of the payment).

Thank you!

**Comment by dcworldwide** [comment]
Use PP Payouts and move a portion as a fee payment to an account you own.

**Comment by vincelwt** [comment]
@dcworldwide Just to be sure using Payouts means we are the merchant of record and have to handle refund, disputes, etc... right? Also it seems there is fee with using Payout so the total (initial payment fee + payout fee) would end up being quite a lot

**Comment by dcworldwide** [comment]
Yes. What other choice do you have..

**Comment by jcorkhill** [comment]
@dcworldwide @vincelwt 

One option I've pondered is to use the Payments/Express Checkout API as to send funds from one user to another (for a Marketplace like app), and then have the seller pay me (again through the Payments API) for a shipping label. 

Although you end up getting charged twice and although it's somewhat of a nasty flow and a bad UX, you do get Purchase Protection between buyers and sellers so that you don't have to handle disputes.
